core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +314,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput' &&
+                  maxSelectionCount === 1) {
+        var numChoices = $scope.getAnswerChoices().length;
+        choiceRules = [];
+        for (var i = 0; i < answerGroups.length; i++) {
+          choiceRules.push(false);
+        }
+
+        var newChoiceRules = function(current) {
+          return choiceRules.map(function(element, index) {
+            if (index !== current || element === true) {
+              return true;
+            } else if (index === current && element !== true) {
+              return false;
+            }
+          });
+        };
+
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              // If the rule_type of the first input is equalTo or containsOne,
+              // then set first choiceRules to true
+              if (ruleSpecs[j].rule_type === 'Equals') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type === 'ContainsAtLeastOneOf') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type ===
+                'DoesNotContainAtLeastOneOf') {
+                // If the ruleType is omits one of, then set the others to true
+                var current = i;
+                choiceRules = newChoiceRules(current);
```

Comment:
Right. As I stated above, I had found previously that I still required both checks to get an accurate view of whether an answer choice had been handled. With the extra validation, this may not be true anymore so I'll test.